File: BridgeX/app/src/main/java/cz/lriedel/bridgex/CustomWebViewClient.java
package cz.lriedel.bridgex;

import android.content.Intent;
import android.webkit.WebResourceRequest;
import android.webkit.WebView;
import android.webkit.WebViewClient;

public class CustomWebViewClient extends WebViewClient {

    private final MainActivity mainActivity;

    public CustomWebViewClient(MainActivity mainActivity) {
        this.mainActivity = mainActivity;
    }

    @Override
    public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
        if (request.getUrl().toString().startsWith(cz.lriedel.bridgex.BuildConfig.PORTAL_BASE_URL)) {
            return false;
        }

        Intent intent = new Intent(Intent.ACTION_VIEW, request.getUrl());
        mainActivity.startActivity(intent);
        return true;
    }
}

File: BridgeX/app/src/main/java/cz/lriedel/bridgex/ui/CustomWebViewClient.kt
package cz.lriedel.bridgex.ui

import android.content.Intent
import android.webkit.WebResourceRequest
import android.webkit.WebView
import android.webkit.WebViewClient
import cz.lriedel.bridgex.BuildConfig
import cz.lriedel.bridgex.MainActivity

class CustomWebViewClient(
    private val mainActivity: MainActivity
) : WebViewClient() {
    override fun shouldOverrideUrlLoading(view: WebView, request: WebResourceRequest): Boolean {
        if (request.url.toString().startsWith(BuildConfig.PORTAL_BASE_URL)) {
            return false
        }

        val intent = Intent(Intent.ACTION_VIEW, request.url)
        mainActivity.startActivity(intent)
        return true
    }
}
